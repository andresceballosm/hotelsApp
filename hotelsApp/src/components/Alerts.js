import { Alert } from 'react-native';

export const showAlertError = (message) => {
    setTimeout( () => {	
        Alert.alert(
            'Error!',
            message,
            [
                {   text: 'ok', onPress: () =>  console.log('error')},
            ],
            { cancelable: false },   
        );
    },600)
}

export const showAlertSuccess = (message) => {
    setTimeout( () => {	
        Alert.alert(
            'Info!',
            message,
            [
                {   text: 'ok', onPress: () =>  console.log('error')},
            ],
            { cancelable: false },   
        );
    },600)
}