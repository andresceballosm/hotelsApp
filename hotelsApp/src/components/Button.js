import React from 'react'
import { Text, TouchableOpacity, StyleSheet, Image } from 'react-native'

export const ButtonGeneral = (props) => {
    return (
        <TouchableOpacity 
        activeOpacity={ props.invalid ? 1 : 0.3 } 
        disabled={props.invalid} 
        style={[styles.button, { backgroundColor: props.color, width: props.width ? props.width : 200}]}
        onPress={()=>{ props.click()}} >
            <Text style={{color:props.fontColor, fontSize:16}}>{props.title}</Text>
        </TouchableOpacity>
    )
};

export const ButtonBackDown = (props) => {
    return (
        <TouchableOpacity 
        onPress={()=>{ props.navigation.pop()}}
        style={styles.navigationHeaderContainer}>
            <Image style={props.imageStyle}
            source={require('../../assets/icons/down.png')}/>
        </TouchableOpacity>
    )
}

const styles = StyleSheet.create({
    button: {
        alignItems:'center',
        height:45,
        borderRadius:20,
        justifyContent:'center',
        width:200
    },
    buttonSombra: {
        alignItems:'center',
        height:45,
        borderRadius:20,
        justifyContent:'center',
        width:200,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 3,
        },
        shadowOpacity: 0.29,
        shadowRadius: 4.65,

        elevation: 7,
        borderRadius:10
    },
})