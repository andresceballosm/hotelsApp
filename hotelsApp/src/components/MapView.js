import React from 'react';
import MapView from 'react-native-maps';

const MapsView = (props) => {
  return (
      <MapView
          style={{ flex: 1 }}
          region={props.region}
          >
         <MapView.Marker
            draggable
            coordinate={props.region} /> 
      </MapView>
  )
}

export default MapsView;