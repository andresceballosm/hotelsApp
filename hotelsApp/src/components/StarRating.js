import React, { Component } from 'react';
import {
	StyleSheet,
	View,
	Image
} from 'react-native';

export default class StarRating extends Component {
	render() {
        let ratings = this.props.ratings;
        let stars = [];

        for (var i = 1; i <= 5; i++) {
			let path = require('../../assets/icons/star-gold.png');
			if (i > ratings) {
				path = require('../../assets/icons/star-white.png');
			}

			stars.push((<Image key={i} style={styles.image} source={path} />));
        }
        
		return (
			<View style={ styles.container }>
              { stars }
			</View>
		);
	}
}

const styles = StyleSheet.create({
    container: {
		flexDirection: 'row',
		alignItems: 'center'
	},
	image: {
		width: 25,
		height: 25
	},
});