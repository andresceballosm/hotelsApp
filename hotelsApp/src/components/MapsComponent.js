import React from 'react';
import { View } from 'react-native'
import MapView from 'react-native-maps'

export const MapsComponent = (props) => {
    const latitude = props.latitude;
    const longitude = props.longitude;
    const region = {
      latitude,
      longitude,
      latitudeDelta: 0.005,
      longitudeDelta: 0.005
    }
    return (
        <MapView
          style = {{ flex:1 }}
          region={region}
        >
          <MapView.Marker
            title={props.hotelName}
            coordinate={region} />
        </MapView>  
    );
}
  