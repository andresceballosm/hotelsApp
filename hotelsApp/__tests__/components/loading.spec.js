import 'react-native';
import React from 'react' // Importante cargar React antes que react-test-renderer
import renderer from 'react-test-renderer';
import { LoadingSmall } from '../../src/components/LoadingSmall';

it('renders correctly', () => {
  const loading = renderer.create(
    <LoadingSmall />
  ).toJSON();
  
  expect(loading).toMatchSnapshot();
});