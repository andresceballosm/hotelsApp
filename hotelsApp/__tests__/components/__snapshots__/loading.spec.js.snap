// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000040",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 10,
          "display": "flex",
          "height": 50,
          "justifyContent": "space-around",
          "width": 50,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</Modal>
`;
